### PR TITLE
Add support for publishing raw IMU data in AP_Periph

### PR DESCRIPTION
**Pull request by bugobliterator**
* Also fixed build with both scripting and GCS enabled
* Changes name from CubePilot-CANMod and -PPPGW to CubeNode

### CODE REVIEW

File: Tools/AP_Periph/imu.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#include <dronecan_msgs.h>
+
+/*
+  update CAN magnetometer
+ */
+void AP_Periph_FW::can_imu_update(void)
+{
+#ifdef HAL_PERIPH_ENABLE_IMU
+    while (true) {
+        imu.update();
```
Comment (tridge) [inline Tools/AP_Periph/imu.cpp:11]: need a 100usec at least sleep here

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/defaults_bootloader.h [hunk 2]
```diff
@@ -15,11 +15,6 @@
 #define HAL_GCS_ENABLED 0
 #endif
 
-// by default bootloaders don't use INS:
-#ifndef AP_INERTIALSENSOR_ENABLED
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/scripts/defaults_bootloader.h:19]: need to test that no bootloader except the CubeNode one changes with this change
so build pre this PR, mv the build to build.orig, re-build PR, compare the bin files with dirdiff or similar to ensure no changes

**Review by tridge — CHANGES_REQUESTED** [review]
@peterbarker I'd like you to check the header changes if you can, thanks!

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 3]
```diff
@@ -854,9 +854,11 @@ void AP_InertialSensor::_start_backends()
         _backends[i]->start();
     }
 
+#ifndef HAL_BUILD_AP_PERIPH
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp:857]:
```suggestion
#if AP_INERTIALSENSOR_ALLOW_NO_SENSORS
```

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp [hunk 4]
```diff
@@ -1324,8 +1326,10 @@ AP_InertialSensor::detect_backends(void)
         #else
         DEV_PRINTF("INS: unable to initialise driver\n");
         GCS_SEND_TEXT(MAV_SEVERITY_DEBUG, "INS: unable to initialise driver");
+        #ifndef HAL_BUILD_AP_PERIPH
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor.cpp:1329]:
```suggestion
        #if !AP_INERTIALSENSOR_ALLOW_NO_SENSORS
```

File: libraries/AP_InertialSensor/AP_InertialSensor_config.h [hunk 5]
```diff
@@ -48,7 +48,7 @@
 #endif
 
 #ifndef AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED
-#define AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED AP_INERTIALSENSOR_ENABLED
+#define AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED (AP_INERTIALSENSOR_ENABLED && !defined(HAL_BUILD_AP_PERIPH))
```
Comment (peterbarker) [inline libraries/AP_InertialSensor/AP_InertialSensor_config.h:51]: set ` AP_INERTIALSENSOR_HARMONICNOTCH_ENABLED` in defaults header file instead

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/scripts/defaults_bootloader.h (hunk 2 above)
Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/hwdef/scripts/defaults_bootloader.h:19]: @peterbarker has tested, all good

File: Tools/AP_Periph/imu.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
+#include "AP_Periph.h"
+#include <dronecan_msgs.h>
+
+/*
+  update CAN magnetometer
+ */
+void AP_Periph_FW::can_imu_update(void)
+{
+#ifdef HAL_PERIPH_ENABLE_IMU
```
[the remark this answers is not in the document]
Comment (bugobliterator) [reply Tools/AP_Periph/imu.cpp]: done

File: libraries/AP_InertialSensor/AP_InertialSensor.cpp (hunk 3 above)
Comment (bugobliterator) [reply libraries/AP_InertialSensor/AP_InertialSensor.cpp]: changed

File: libraries/AP_InertialSensor/AP_InertialSensor_config.h (hunk 5 above)
Comment (bugobliterator) [reply libraries/AP_InertialSensor/AP_InertialSensor_config.h]: changed

**Review by tridge — APPROVED** [review]

**Pull request merged**